constraint-solver/src/solver/base.rs

```diff
@@ -478,6 +482,27 @@ where
         exprs
     }
 
+    fn inline_affine(&mut self) -> bool {
+        let mut progress = false;
+        let affine_equivalences = self
+            .constraint_system
+            .system()
+            .algebraic_constraints()
+            .iter()
+            .filter(|constr| {
+                constr.expression.is_affine() && constr.referenced_unknown_variables().count() == 2
```

Comment:
We should filter out constraints that contain new variables, although internally it could make sense.